jupyterlab_dash/__init__.py

```diff
@@ -33,6 +34,13 @@ def run(*args, **kwargs):
             # Serve App
             sys.stdout = self.stderr_queue
             sys.stderr = self.stderr_queue
+
+            # Set pathname prefix for jupyter-server-proxy
+            url = next(notebookapp.list_running_servers())['url']
+
+            path = urlparse(url).path
+            app.config.update({'requests_pathname_prefix': f'{path}proxy/{self.port}/'})
```

Comment:
@jonmmease I'd appreciate your feedback on this. I'm getting the first running Jupyter server, parsing out the path, and using that to set the dash server's `request_pathname_prefix` (which gets prepended to each outbound request). I'm not sure how brittle this approach may be. We are also getting this _base pathname_ on the client using jupyterlab's `PageConfig.getBaseUrl`, so another approach would be to send this base pathname from the client to the kernel.

Reply:
Thanks for calling out this decision.  I am a bit concerned that this could fail in a JupyterHub context where there may be multiple notebook servers running simultaneously.  Would you mind taking a stab at sending this base pathname from the client back to the kernel?

Reply:
@gnestor Is this something you're interested in trying out?  If not I may be able to take a stab at it over the weekend.

Reply:
I'm going to be skiing this weekend so go ahead. I think that sending the base pathname from the client makes the most sense. If you don't get to it this weekend, I'll work on it next week.